ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivateTransactionSimulator.java

```diff
@@ -0,0 +1,203 @@
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpec;
+import org.hyperledger.besu.ethereum.mainnet.TransactionValidator;
+import org.hyperledger.besu.ethereum.mainnet.ValidationResult;
+import org.hyperledger.besu.ethereum.transaction.CallParameter;
+import org.hyperledger.besu.ethereum.trie.MerklePatriciaTrie;
+import org.hyperledger.besu.ethereum.vm.BlockHashLookup;
+import org.hyperledger.besu.ethereum.vm.DebugOperationTracer;
+import org.hyperledger.besu.ethereum.worldstate.WorldStateArchive;
+
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes;
+
+/*
+ * Used to process transactions for priv_call.
+ *
+ * The processing won't affect the private world state, it is used to execute read operations on the
+ * blockchain.
+ */
+public class PrivateTransactionSimulator {
+
+  private static final Hash EMPTY_ROOT_HASH = Hash.wrap(MerklePatriciaTrie.EMPTY_TRIE_NODE_HASH);
+
+  // Dummy signature for transactions to not fail being processed.
+  private static final SECP256K1.Signature FAKE_SIGNATURE =
+      SECP256K1.Signature.create(SECP256K1.HALF_CURVE_ORDER, SECP256K1.HALF_CURVE_ORDER, (byte) 0);
+
+  private static final Address DEFAULT_FROM =
+      Address.fromHexString("0x0000000000000000000000000000000000000000");
+
+  private final Blockchain blockchain;
+  private final WorldStateArchive worldStateArchive;
+  private final ProtocolSchedule<?> protocolSchedule;
+  private final PrivacyParameters privacyParameters;
+
+  public PrivateTransactionSimulator(
+      final Blockchain blockchain,
+      final WorldStateArchive worldStateArchive,
+      final ProtocolSchedule<?> protocolSchedule,
+      final PrivacyParameters privacyParameters) {
+    this.blockchain = blockchain;
+    this.worldStateArchive = worldStateArchive;
+    this.protocolSchedule = protocolSchedule;
+    this.privacyParameters = privacyParameters;
+  }
+
+  public Optional<PrivateTransactionProcessor.Result> process(
+      final String privacyGroupId,
+      final String enclaveKey,
+      final CallParameter callParams,
+      final Hash blockHeaderHash) {
+    final BlockHeader header = blockchain.getBlockHeader(blockHeaderHash).orElse(null);
+    return process(privacyGroupId, enclaveKey, callParams, header);
+  }
+
+  public Optional<PrivateTransactionProcessor.Result> process(
+      final String privacyGroupId,
+      final String enclaveKey,
+      final CallParameter callParams,
+      final long blockNumber) {
+    final BlockHeader header = blockchain.getBlockHeader(blockNumber).orElse(null);
+    return process(privacyGroupId, enclaveKey, callParams, header);
+  }
+
+  public Optional<PrivateTransactionProcessor.Result> processAtHead(
+      final String privacyGroupId, final String enclaveKey, final CallParameter callParams) {
+    return process(privacyGroupId, enclaveKey, callParams, blockchain.getChainHeadHeader());
+  }
```

Comment:
Looks like this method isn't being used. If this is the case we can delete it.

Reply:
Deleted.